### PR TITLE
Init embeddings

### PR DESCRIPTION
**Pull request by hinthornw**

### CODE REVIEW

File: libs/langchain/langchain/embeddings/base.py [hunk 1]
```diff
@@ -1,4 +1,226 @@
...
+        return HuggingFaceEmbeddings(model_name=model_name, **kwargs)
+    else:
+        raise ValueError(
+            f"Provider '{provider}' is not supported.\n"
+            f"Supported providers and their required packages:\n"
+            f"{_get_provider_list()}"
+        )
+
+
+@functools.lru_cache
```
Comment (baskaryan) [inline libs/langchain/langchain/embeddings/base.py:153]: should we set max size

File: libs/langchain/langchain/embeddings/base.py [hunk 2]
```diff
@@ -1,4 +1,226 @@
...
+def _check_pkg(pkg: str) -> None:
+    """Check if a package is installed."""
+    if not util.find_spec(pkg):
+        raise ImportError(
+            f"Could not import {pkg} python package. "
+            f"Please install it with `pip install {pkg}`"
+        )
+
+
+def embedding_model(
```
Comment (baskaryan) [inline libs/langchain/langchain/embeddings/base.py:163]: in a vacuum i do like this name more but i wonder if we should stick to init_embedding_model for consistency

File: libs/langchain/langchain/embeddings/base.py [hunk 3]
```diff
@@ -1,4 +1,226 @@
...
+    Raises:
+        ValueError: If the model provider is not supported or cannot be determined
+        ImportError: If the required provider package is not installed
+
+    Examples:
+        >>> # Using a model string
+        >>> model = init_embedding_model("openai:text-embedding-3-small")
+        >>> model.embed_query("Hello, world!")
+        
+        >>> # Using explicit provider
```
Comment (baskaryan) [inline libs/langchain/langchain/embeddings/base.py:200]: would double check that these render correctly in api ref, not sure they will

File: libs/langchain/langchain/embeddings/base.py (hunk 2 above)
Comment (baskaryan) [reply libs/langchain/langchain/embeddings/base.py:163]: also might want to mark as beta initially

**Review by baskaryan — COMMENTED** [review]
slick

File: libs/langchain/langchain/embeddings/base.py (hunk 2 above)
Comment (hinthornw) [reply libs/langchain/langchain/embeddings/base.py:163]: ya i'm fine with it

File: libs/langchain/langchain/embeddings/base.py (hunk 1 above)
Comment (hinthornw) [reply libs/langchain/langchain/embeddings/base.py:153]: sure! though I hope people don't spam the provider list

File: libs/langchain/langchain/embeddings/base.py (hunk 3 above)
Comment (hinthornw) [reply libs/langchain/langchain/embeddings/base.py]: wlil fix

File: libs/langchain/langchain/embeddings/base.py [hunk 4]
```diff
@@ -1,4 +1,235 @@
...
+    .. versionadded:: 0.3.9
+    """
+    if not model:
+        providers = _SUPPORTED_PROVIDERS.keys()
+        raise ValueError(
+            "Must specify model name. "
+            f"Supported providers are: {', '.join(providers)}"
+        )
+
+    return _init_embeddings_helper(model, provider=provider, **kwargs)
```
Comment (baskaryan) [inline libs/langchain/langchain/embeddings/base.py:229]: ubernit, helper doesn't really feel needed here

**Review by baskaryan — APPROVED** [review]

**Pull request merged**